staging/src/k8s.io/apiserver/pkg/cel/environment/base.go

```diff
@@ -149,13 +152,34 @@ var baseOptsWithoutStrictCost = []VersionedOptions{
 	// Authz selectors
 	{
 		IntroducedVersion: version.MajorMinor(1, 31),
-		// TODO(liggitt): limit this to feature enablement
+		FeatureEnabled: func() bool {
+			enabled := utilfeature.DefaultFeatureGate.Enabled(genericfeatures.AuthorizeWithSelectors)
+			authzSelectorsLibraryInit.Do(func() {
+				// Record the first time feature enablement was checked for this library.
+				// This is checked from integration tests to ensure no cached cel envs
+				// are constructed before feature enablement is effectively set.
+				authzSelectorsLibraryEnabled.Store(enabled)
+				// Uncomment to debug where the first initialization is coming from if needed.
+				// debug.PrintStack()
+			})
```

Comment:
this is solely for the benefit of the integration tests making sure feature gates get parsed before AuthzSelector ever has to check for enablement